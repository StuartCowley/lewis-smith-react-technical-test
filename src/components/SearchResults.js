import React from "react";
import "../styles/searchresults.css";

const SearchResults = ({ results }) => {
  if (!results.length) {
    return <p className="card-no-result" style={({ textAlign: "center", fontWeight: "bold", fontSize: "20px" })}>No results</p>;
  } else {
    return (
      <div className="card-image-parent">
        {results.map((image, index) => (
          <img className="card-image" src={image} alt="space" key={index} />
        ))}
      </div>
    );
  }
};

export default SearchResults;
