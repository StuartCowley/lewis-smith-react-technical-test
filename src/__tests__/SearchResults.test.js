import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  it("renders correctly", () => {
    const results = [];
    const { asFragment } = render(<SearchResults results={results} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly with results", () => {
    const results = [
      "https:/example.com/image1.jpg",
      "https:/example.com/image2.jpg",
      "https:/example.com/image3.jpg",
    ];

    const { asFragment } = render(<SearchResults results={results} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders no results message when there are no results", () => {
    const results = [];
    render(<SearchResults results={results} />);
    const noResultsMessage = screen.getByText("No results");

    expect(noResultsMessage).toBeInTheDocument();
  });

  it("renders images when results is not empty", () => {
    const results = [
      "https:/example.com/image1.jpg",
      "https:/example.com/image2.jpg",
      "https:/example.com/image3.jpg",
    ];
    render(<SearchResults results={results} />);
    const images = screen.queryAllByAltText("space");

    expect(images.length).toBe(3);
  });
});
