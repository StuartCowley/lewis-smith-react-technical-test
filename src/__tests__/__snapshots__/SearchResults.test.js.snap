// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <p>
    No results
  </p>
</DocumentFragment>
`;

exports[`SearchResults renders correctly with results 1`] = `
<DocumentFragment>
  <img
    alt="space"
    class="card-image"
    src="https:/example.com/image1.jpg"
  />
  <img
    alt="space"
    class="card-image"
    src="https:/example.com/image2.jpg"
  />
  <img
    alt="space"
    class="card-image"
    src="https:/example.com/image3.jpg"
  />
</DocumentFragment>
`;
